src/components/application_manager/test/rpc_service_impl_test.cc

```diff
@@ -0,0 +1,832 @@
+}
+
+TEST_F(RPCServiceImplTest, Handle_HappyPathWithClosedSession) {
+  auto message_handler = std::make_shared<rpc_service::RPCServiceImpl>(
+      mock_app_mngr_,
+      request_controller_,
+      &mock_protocol_handler_,
+      &mock_hmi_handler_,
+      mock_command_holder_,
+      mock_rpc_protection_manager_,
+      hmi_so_factory_,
+      mobile_so_factory_);
+  auto message_to_send = std::make_shared<am::Message>(
+      protocol_handler::MessagePriority::kDefault);
+  message_to_send->set_protocol_version(
+      protocol_handler::MajorProtocolVersion::PROTOCOL_VERSION_1);
+  message_to_send->set_connection_key(kConnectionKey);
+  message_to_send->set_correlation_id(kCorrelationId);
+  message_to_send->set_function_id(kFunctionId);
+  message_to_send->set_message_type(static_cast<am::MessageType>(2));
+  message_to_send->set_json_message(std::string("message"));
+
+  NiceMock<test::components::connection_handler_test::MockConnectionHandler>
+      mock_conn_handler;
+  EXPECT_CALL(mock_app_mngr_, connection_handler())
+      .WillRepeatedly(ReturnRef(mock_conn_handler));
+  ON_CALL(mock_conn_handler, GetConnectionSessionsCount(_))
+      .WillByDefault(Return(2));
+  ON_CALL(mock_protocol_handler_, IsRPCServiceSecure(_))
+      .WillByDefault(Return(true));
+  ON_CALL(mock_app_mngr_, application(_)).WillByDefault(Return(nullptr));
+  EXPECT_CALL(*mock_rpc_protection_manager_, CheckPolicyEncryptionFlag(_, _, _))
+      .WillOnce(Return(true));
+  EXPECT_CALL(mock_protocol_handler_, SendMessageToMobileApp(_, _, _))
+      .WillOnce(Return());
+  EXPECT_CALL(*mock_rpc_protection_manager_,
+              RemoveFromEncryptionNeededCache(_, _))
+      .WillOnce(Return());
+  EXPECT_CALL(mock_conn_handler, CloseSession(_, _)).WillOnce(Return());
+
+  message_handler->Handle(
+      am::rpc_service::impl::MessageToMobile(message_to_send, true));
+}
+
+TEST_F(RPCServiceImplTest, SendMessageToMobile_HappyPath) {
+  using namespace application_manager::plugin_manager::plugin_names;
+  namespace jhs = ns_smart_device_link::ns_json_handler::strings;
+  auto func_id = mobile_apis::FunctionID::OnSystemRequestID;
+  auto source = am::commands::Command::CommandSource::SOURCE_SDL;
+
+  auto message = CreateMessage(smart_objects::SmartType_Map);
+  (*message)[am::strings::params][am::strings::connection_key] = kConnectionKey;
+  (*message)[am::strings::params][am::strings::function_id] = func_id;
+  (*message)[am::strings::msg_params][am::strings::request_type] =
+      mobile_apis::RequestType::PROPRIETARY;
+  (*message)[am::strings::msg_params][am::strings::result_code] =
+      mobile_apis::Result::SUCCESS;
+
+  (*message)[jhs::S_PARAMS][jhs::S_FUNCTION_ID] = func_id;
+  (*message)[jhs::S_PARAMS][jhs::S_MESSAGE_TYPE] =
+      mobile_apis::messageType::request;
+  (*message)[jhs::S_PARAMS][jhs::S_PROTOCOL_TYPE] = mobile_protocol_type_;
+  (*message)[jhs::S_PARAMS][jhs::S_PROTOCOL_VERSION] = 1;
```

Comment:
Don't use 'magic numbers', use a constant here and declare it in anonymous namespace at the top of file.
Please check all similar places across file.

Reply:
Done in commit d5ec8240cfac58e5ea9f7f9ed13f8127a02f94b9

Reply:
Done in commit d5ec824